### DISCUSSION TITLE
[Feature Request] Add commit being operated on as an arg to 'git test'

### DISCUSSION THREAD

**Discussion by mfulbright** [arxanas/git-branchless#1346] [Q&A] [answered]
So far as I know, there is no way to refer to the commit that a `git test` command is operating on. I would find it useful to be able to do this:

\# Fail any commits that still have "TODO_BEFORE_SUBMIT" in their changes:
$ git test run --exec '[[ ! $(git diff "$1^" "$1" | grep TODO_BEFORE_SUBMIT) ]]'

`$1` represents the commit that `git test` is currently operating on, e.g. if `git test` is going to operate on 2 commits, `abc` and `def`, then the first test would be
`[[ ! $(git diff "abc^" "abc" | grep TODO_BEFORE_SUBMIT) ]]`
and the second test would be
`[[ ! $(git diff "def^" "def" | grep TODO_BEFORE_SUBMIT) ]]`

**Comment by claytonrcarter** [comment, marked as answer]
Hi there. It looks like this may already be supported via the `$BRANCHLESS_TEST_COMMIT` env var. (I didn't see that mentioned in the wiki docs, but it looks like it's being injected in the code: https://github.com/arxanas/git-branchless/blob/master/git-branchless-test/src/lib.rs#L2645) Can you try this and see if it works for you:
```
# Fail any commits that still have "TODO_BEFORE_SUBMIT" in their changes:
$ git test run --exec '[[ ! $(git diff "$BRANCHLESS_TEST_COMMIT"^ "$BRANCHLESS_TEST_COMMIT" | grep TODO_BEFORE_SUBMIT) ]]'
```

Although in this case, I wonder if you could bypass the env var entirely and just do `git diff @^` to get the diff for the current commit and its parent. 

Fair warning: I didn't test either of these, but maybe they're worth a shot.

**Reply by arxanas** [reply]
I think `HEAD`/`@` is equivalent to `BRANCHLESS_TEST_COMMIT` in all cases currently. Having it in an environment variable could be useful for the case of a fix command that doesn't materialize a working copy.

I updated the Wiki page to include the environment variables: https://github.com/arxanas/git-branchless/wiki/Command:-git-test#environment-variables

**Reply by mfulbright** [reply]
Hi - yeah, that worked great! Thanks claytoncarter for the answer, and thanks arxanas for documenting it. Appreciate it!
